tests/infra/network.py

```diff
@@ -695,14 +695,17 @@ def find_primary(self, timeout=3, log_capture=None):
                     try:
                         logs = []
                         res = c.get("/node/network", log_capture=logs)
-                        if res.status_code != 200:
-                            continue
+                        assert res.status_code == http.HTTPStatus.OK.value, res
```

Comment:
The `/node/network` endpoint isn't guaranteed to return `200` but at this stage, we only loop through nodes that have joined the service so this assert is actually safe.